src/gui/sharelinkwidget.cpp

```diff
@@ -539,24 +562,37 @@ void ShareLinkWidget::slotLinkContextMenuActionTriggered(QAction *action)
 {
     const auto state = action->isChecked();
     SharePermissions perm = SharePermissionRead;
-
+    _ui->currentPermission_3->setElideMode(Qt::ElideRight);
     if (action == _addAnotherLinkAction.data()) {
         emit createLinkShare();
 
     } else if (action == _readOnlyLinkAction && state) {
         _linkShare->setPermissions(perm);
+        if(_isFile)
+        {
+            _ui->currentPermission_3->setEnabled(false);
+        }
+        else
+        {
+            _ui->currentPermission_3->setEnabled(true);
+        }
```

Comment:
This does not fit the style of other conditional statements throughtout the desktop client (see the parent `if` statements for instance)